src/Bicep.LangServer/Handlers/BicepDecompileCommandHandler.cs

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.ServiceModel.Channels;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Deployments.Core.Entities;
+using Azure.Deployments.Core.Helpers;
+using Azure.ResourceManager.Resources.Models;
+using Bicep.Core.Analyzers.Linter;
+using Bicep.Core.Analyzers.Linter.ApiVersions;
+using Bicep.Core.Configuration;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Emit;
+using Bicep.Core.Features;
+using Bicep.Core.FileSystem;
+using Bicep.Core.Registry;
+using Bicep.Core.Semantics;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.Workspaces;
+using Bicep.Decompiler;
+using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Providers;
+using Bicep.LanguageServer.Telemetry;
+using Bicep.LanguageServer.Utils;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.ResourceStack.Common.Extensions;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using OmniSharp.Extensions.LanguageServer.Protocol.Window;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+using SharpYaml.Serialization.Logging;
+using static Bicep.LanguageServer.Handlers.BicepDecompileCommandHandler;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    public enum BicepDecompileCommandStatus
+    {
+        Canceled = 1,
+        Failed = 2,
+        Success = 3,
+    }
+
+    public record BicepDecompileCommandParams(DocumentUri jsonUri);
+    public record BicepDecompileCommandResult(BicepDecompileCommandStatus status, string output, Uri? bicepUri, string? failureError);
+
+    /// <summary>
+    /// Handles a request from the client to decompile a JSON file for given a file path, creating one or more bicep files
+    /// </summary>
+    /// <remarks>
+    /// Using ExecuteTypedResponseCommandHandlerBase instead of IJsonRpcRequestHandler because IJsonRpcRequestHandler will throw "Content modified" if text changes are detected, and for this command
+    /// that is expected.
```

Comment:
Still some "content modifed" in some scenarios, will fix in separate PR

Reply:
Decided to try disabling contentModified support, but waiting for next release, therefore blocked by this.